subworkflows/local/align_cellrangermulti.nf

```diff
@@ -0,0 +1,247 @@
+            // parse it to generate the cmo / frna samplesheets to be used by each sample.
+            //
+            // Here, to guarantee it and take advantage of the "FIFO"-rule and are sure that the data used in the
+            // module is from the same sample from the "normal" samplesheet. We have to use the .concat().groupTuple()
+            // pipe instead of .join() because .join() outputs first the arrays that could be joined and afterwards
+            // the ones with "remainders", thus, we would not ensure "FIFO" and the same order.
+            //
+            // To guarantee this, we can define two nf-tests, one having only one sample with CMO and another with two
+            // samples using CMOs, even if wrongly/repeated, but just to guarantee FIFO is working.
+            //
+
+            PARSE_CELLRANGERMULTI_SAMPLESHEET( ch_multi_samplesheet )
+
+            ch_grouped_fastq.gex
+            .map{ [it[0].id] }
+            .concat( PARSE_CELLRANGERMULTI_SAMPLESHEET.out.cmo.flatten().map { [ "${it.baseName}" - "_cmo", it ] } )
+            .groupTuple()
+            .map { if ( it.size() == 2 ) { it[1] } else { empty_file } } // a correct tuple from snippet will have: [ sample, cmo.csv ]
+            .flatten()
+            .set { ch_cmo_barcode_csv }
+
+            ch_grouped_fastq.gex
+            .map{ [it[0].id] }
+            .concat( PARSE_CELLRANGERMULTI_SAMPLESHEET.out.frna.flatten().map { [ "${it.baseName}" - "_frna", it ] } )
+            .groupTuple()
+            .map { if ( it.size() == 2 ) { it[1] } else { empty_file } } // a correct tuple from snippet will have: [ sample, frna.csv ]
+            .flatten()
+            .set { ch_frna_sample_csv }
+
+        } else {
+            ch_cmo_barcode_csv = empty_file
+            ch_frna_sample_csv = empty_file
+        }
+
+        //
+        // Prepare GTF
+        //
+        if (!cellranger_gex_index || !cellranger_vdj_index) {
+
+            // Filter GTF based on gene biotypes passed in params.modules
+            CELLRANGER_MKGTF ( ch_gtf )
+            ch_versions = ch_versions.mix(CELLRANGER_MKGTF.out.versions)
+
+        }
+
+        //
+        // Prepare gex reference (Normal Ref)
+        //
+        if ( !cellranger_gex_index ) {
+
+            // Make reference genome
+            CELLRANGER_MKREF(
+                ch_fasta,
+                CELLRANGER_MKGTF.out.gtf,
+                "gex_reference"
+            )
+            ch_versions = ch_versions.mix(CELLRANGER_MKREF.out.versions)
+            ch_cellranger_gex_index = CELLRANGER_MKREF.out.reference.ifEmpty { empty_file }
+
+        } else {
+            ch_cellranger_gex_index = cellranger_gex_index
+        }
+
+        //
+        // Prepare vdj reference (Special)
+        //
+        if ( !cellranger_vdj_index ) {
+
+            if ( !params.skip_cellrangermulti_vdjref  ) { // if user uses cellranger multi but does not have VDJ data
```

Comment:
Would be nice to autodetect this based on the feature types specified in the samplesheet. 

But I currently don't have any idea how to do this properly (The info we need is in the `ch_grouped_fastq.vdj` channel - but we can't have an `if` based on a Queue channel).